shell/browser/native_window_mac.mm

```diff
@@ -114,6 +114,43 @@ static bool FromV8(v8::Isolate* isolate,
 
 namespace electron {
 
+class NativeAppWindowFrameViewMacClient
+    : public views::NativeFrameViewMacClient {
+ public:
+  NativeAppWindowFrameViewMacClient(views::Widget* frame,
+                                    NativeWindowMac* window)
+      : frame_(frame), native_app_window_(window) {}
+
+  NativeAppWindowFrameViewMacClient(const NativeAppWindowFrameViewMacClient&) =
+      delete;
+  NativeAppWindowFrameViewMacClient& operator=(
+      const NativeAppWindowFrameViewMacClient&) = delete;
+
+  ~NativeAppWindowFrameViewMacClient() override = default;
+
+  std::optional<int> NonClientHitTest(const gfx::Point& point) override {
+    if (frame_->IsFullscreen()) {
+      return HTCLIENT;
+    }
+
+    // Check for possible draggable region in the client area for the frameless
+    // window.
+    int contents_hit_test = native_app_window_->NonClientHitTest(point);
+    if (contents_hit_test != HTNOWHERE)
+      return contents_hit_test;
+
+    return HTCLIENT;
+  }
+
+ private:
+  const raw_ptr<views::Widget> frame_;
+  // Weak. Owned by extensions::AppWindow (which manages our Widget via its
+  // WebContents).
+  const raw_ptr<NativeWindowMac, DanglingUntriaged> native_app_window_;
+};
+
+NativeWindowMac::~NativeWindowMac() = default;
```

Comment:
Don't think the destructor needs to be moved above the constructor, is that by accident ?